idioms/rustdoc-init.md

```diff
@@ -0,0 +1,83 @@
+# Easy doc initialization
+
+## Description
+
+If a struct takes significant effort to initialize, when writing docs it can be quicker to wrap your example with a 
+function the struct as an argument.
+
+## Motivation
+Sometimes there is a struct with multiple or complicated parameters and several methods.
+Each of these methods should have examples. 
+
+For example:
+
+```rust
+
+struct Connection {
+    name: String,
+    stream: TcpStream,
+    //...
+}
+
+impl Connection {
+    /// Sends a request over the connection.
+    /// # Example
+    /// ```no_run
+    /// # //This is some really borring boiler plate to get an example working.
+    /// # let stream = TcpStream::connect("127.0.0.1:34254");
+    /// # let connection = Connection{ name: "Foo".to_owned(), stream: stream };
+    /// # let request = Request::new("ReuqestId", RequestType::Get, "dummy_payload");
+    /// let result = connection.send_reqest(request);
+    /// // do stuff with result.
+    /// ```
+    fn send_reqest(&self, request: Request) -> Result<Status, SendErr> {
+        //...
+    }
+        
+    /// Oh no all that boiler plate needs to be repeated here !!!
+    fn check_status(&self) -> Status {
+        //...
+    }
+}
+```
+
+## Example
+Instead of typing all of this boiler plate to create an `Connection` and `Request` it is easier to just create a wrapping dummy function which takes them as arguments:
+```rust
+
+struct Connection {
+    name: String,
+    stream: TcpStream,
+    //...
+}
+
+impl Connection {
+    /// Sends a request over the connection.
+    /// # Example
+    /// ```
+    /// # fn call_send(connection: Connection, request: Request) {
+    /// let result = connection.send_reqest();
+    /// // do stuff with result.
+    /// # }
+    /// ```
+    fn send_reqest(&self, request: Request) {
+        //...
```

Comment:
```suggestion
    /// # fn call_send(connection: Connection, request: Request) {
    /// let response = connection.send_request();
    /// assert!(response.is_ok());
    /// # }
    /// ```
    fn send_request(&self, request: Request) {
        // ...
```

Reply:
This should say "run" rather than "runned"

Reply:
I incorporated this, but moved the comment below so it wouldn't look part of the doc for `send_request`